template/public/app.js

```diff
@@ -1,34 +1,48 @@
-import moment from 'moment';
-import { uiModules } from 'ui/modules';
-import uiRoutes from 'ui/routes';
+import React from "react";
+import moment from "moment";
+import { uiModules } from "ui/modules";
+import chrome from "ui/chrome";
+import uiRoutes from "ui/routes";
+import { render, unmountComponentAtNode } from "react-dom";
 
-import 'ui/autoload/styles';
-import './less/main.less';
-import template from './templates/index.html';
+import "ui/autoload/styles";
+import "./less/main.less";
+import template from "./templates/index.html";
+import { Main } from "./components/main";
 
+const app = uiModules.get("apps/<%= name %>");
+app.config(stateManagementConfigProvider => {
+  stateManagementConfigProvider.disable();
+});
+const controller = ($scope, $route, $interval) => {
+  const elem = document.getElementById("pluginReactRoot");
+  let currentTime = moment($route.current.locals.currentTime);
+  const props = {
+    title: "<%= this.startCase(name) %>",
+    description: "<%= description %>",
+    currentTime: currentTime.format("HH:mm:ss")
+  };
+  render(<Main {...props} />, elem);
+  const stop = $interval(function() {
```

Comment:
And another one here?

```jsx
// Every time the props change, we need to re-mount the React component tree so that it updates the DOM.
currentTime = currentTime.add(1, "second");
render(<Main {...props} currentTime={currentTime.format("HH:mm:ss")} />, elem);
```